src/branchIfEmpty.tsx

```diff
@@ -0,0 +1,30 @@
+import {ReactElement} from 'react'
+import {branch, renderNothing, returns} from 'ad-hok'
+import {isEmpty} from 'lodash'
+import {some} from 'lodash/fp'
+
+import {ensureArray} from './utils'
+
+type BranchIfEmptyType = <TProps extends {}, TPropNames extends keyof TProps>(
+  propNames: Array<TPropNames> | TPropNames,
+  opts?: {
+    returns?: (props: TProps) => ReactElement<any, any> | null
+  },
+) => (
+  props: TProps,
+) => TProps &
+  Required<{[PropName in TPropNames]: NonNullable<TProps[PropName]>}>
```

Comment:
What do you think about supporting path-style prop names in this function (but not the type, since that's impossible)? I think that'd be nice for JS users, and I expected the API to support it (before I thought about the types).